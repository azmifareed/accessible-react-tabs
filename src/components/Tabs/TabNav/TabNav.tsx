import React, { ReactNode } from 'react';
import { Container } from './TabNav.styled';

export interface ITabNavProps {
  children: ReactNode;
  ariaLabel: string;
  onKeyDown: (event: any) => void;
}

export function TabNav ({ children, ariaLabel }: ITabNavProps) {
  return (
    <Container role="tablist" aria-label={ariaLabel}>
      { children }
    </Container>
  );
}
