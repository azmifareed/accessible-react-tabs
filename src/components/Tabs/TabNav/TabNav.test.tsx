import React from 'react';
import { render } from '../../../utils/test-utils';
import { TabNav } from './TabNav';


test('renders TabNav correctly', () => {
  const component = render(<TabNav ariaLabel="test label">test child</TabNav>)
  expect(component.baseElement).toMatchSnapshot();
});


