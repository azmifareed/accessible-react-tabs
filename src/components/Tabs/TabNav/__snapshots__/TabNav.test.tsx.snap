// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TabNav correctly 1`] = `
<body>
  <div>
    <div
      aria-label="test label"
      class="sc-bdnxRM ewiLTn"
      role="tablist"
    >
      test child
    </div>
  </div>
</body>
`;
